aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StorageDownloadTest.java

```diff
@@ -0,0 +1,252 @@
+public final class AWSS3StorageDownloadTest extends StorageInstrumentationTestBase {
+
+    private static final StorageAccessLevel DEFAULT_ACCESS_LEVEL = StorageAccessLevel.PUBLIC;
+    private static final long DEFAULT_TIMEOUT_IN_SECONDS = 10;
+
+    private static final long LARGE_FILE_SIZE = 10 * 1024 * 1024L; // 10 MB
+    private static final long SMALL_FILE_SIZE = 100L;
+
+    private final String largeFileName = "large-test-" + System.currentTimeMillis();
+    private final String smallFileName = "small-test-" + System.currentTimeMillis();
+
+    private File largeFile;
+    private File smallFile;
+
+    private final String destination = "download-test-" + System.currentTimeMillis();
+    private File downloadFile;
+    private StorageDownloadFileOptions options;
+
+    /**
+     * Sets up the required files for testing transfers.
+     * @throws Exception if fails to create temp files
+     */
+    @Before
+    public void setUp() throws Exception {
+        // Set up file to download test-object to
+        downloadFile = createTempFile(destination);
+
+        // Randomly write objects to upload
+        largeFile = createTempFile(largeFileName, LARGE_FILE_SIZE);
+        smallFile = createTempFile(smallFileName, SMALL_FILE_SIZE);
+
+        // Always interact with PUBLIC access for consistency
+        options = StorageDownloadFileOptions.builder()
+                .accessLevel(DEFAULT_ACCESS_LEVEL)
+                .build();
+    }
+
+    /**
+     * Cleans up the S3 bucket for files that were uploaded
+     * during testing processes.
+     */
+    @After
+    public void cleanUp() {
+        String largeFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, largeFileName);
+        String smallFileKey = getS3Key(DEFAULT_ACCESS_LEVEL, smallFileName);
+        cleanUpS3Object(largeFileKey);
+        cleanUpS3Object(smallFileKey);
+    }
+
+    /**
+     * Tests that small file (single-part) downloads successfully.
+     *
+     * @throws Exception if uploading test-file fails
+     */
+    @Test
+    public void testDownloadSmallFile() throws Exception {
+        latchedUpload(smallFile);
+
+        StorageDownloadFileResult result = Await.<StorageDownloadFileResult, StorageException>result(
+                (onResult, onError) ->
+                        Amplify.Storage.downloadFile(
+                                smallFileName,
+                                downloadFile.getAbsolutePath(),
+                                options,
+                                onResult,
+                                onError
+                        )
+        );
+        assertEquals(downloadFile, result.getFile());
+        TestUtils.assertFileEqualsFile(smallFile, downloadFile);
+    }
+
+    /**
+     * Tests that large file (multi-part) downloads successfully.
+     *
+     * @throws Exception if uploading test-file fails
+     */
+    @Test
+    public void testDownloadLargeFile() throws Exception {
+        latchedUpload(largeFile);
```

Comment:
Similar comment here - I think it'd be better to just upload the large file once during setup vs. uploading it here and then also two more times below.

Reply:
yep that's a much better idea